src/insight/systemInsight.go

```diff
@@ -0,0 +1,181 @@
+package insight
+
+import (
+	"errors"
+	"time"
+
+	"github.com/accuknox/auto-policy-discovery/src/libs"
+	ipb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/insight"
+	sys "github.com/accuknox/auto-policy-discovery/src/systempolicy"
+	types "github.com/accuknox/auto-policy-discovery/src/types"
+)
+
+func ConvertSysInsDataToInsResponse(sysdata ipb.SystemInsightData) ipb.InsightResponse {
+	var locsysinsdata ipb.SystemInsightData
+	var insresp ipb.InsightResponse
+
+	locsysinsdata.ContainerName = sysdata.ContainerName
+	locsysinsdata.SysResource = sysdata.SysResource
+
+	insresp.ClusterName = sysdata.ClusterName
+	insresp.NameSpace = sysdata.Namespace
+	insresp.Labels = sysdata.Labels
+	insresp.SystemResource = append(insresp.SystemResource, &locsysinsdata)
+
+	return ipb.InsightResponse{
+		ClusterName:    insresp.ClusterName,
+		NameSpace:      insresp.NameSpace,
+		Labels:         insresp.Labels,
+		SystemResource: insresp.SystemResource,
+	}
+}
+
+func convertWPFSToInsightData(wpfsSet types.ResourceSetMap) types.SysInsightResponseData {
+	var resData types.SysInsightResponseData
+
+	for wpfs, fsset := range wpfsSet {
+		var locFsData types.SystemData
+		var locObsData types.SysInsightData
+
+		// Populate Fileset data(fromsource, process paths and file paths)
+		locFsData.FromSource = wpfs.FromSource
+		if wpfs.SetType == sys.SYS_OP_FILE {
+			locFsData.FilePaths = append(locFsData.FilePaths, fsset...)
+		}
+		if wpfs.SetType == sys.SYS_OP_PROCESS {
+			locFsData.ProcessPaths = append(locFsData.ProcessPaths, fsset...)
+		}
+		if wpfs.SetType == sys.SYS_OP_NETWORK {
+			locFsData.NetworkPaths = append(locFsData.NetworkPaths, fsset...)
+		}
```

Comment:
can use else-if?

Reply:
No... We cannot use else-if here as all 3 cases are possible in same scenario.